From 0e0954c71d09bd09c3c2ffacb7386349c59cbeb8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Elan=20Ruusam=C3=A4e?= <glen@delfi.ee>
Date: Sat, 29 Dec 2012 16:06:20 +0200
Subject: [PATCH] xdg-settings: add MATE support

MATE support was incomplete in b961235b197647d6649ef3d48d7cc2cecafe3d47
it did not add support for xdg-settings which Google Chrome/Chromium
uses to detect and set default browser.

MATE 1.5 uses same schema as gnome3, so shortcut to gnome3 methods

testcases:
$ xdg-settings get default-web-browser
chromium-browser.desktop
$ xdg-settings check
default-web-browser chromium-browser.desktop
yes
---
 scripts/xdg-settings.in | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/scripts/xdg-settings.in b/scripts/xdg-settings.in
index eaed991..b1e43cb 100644
--- a/scripts/xdg-settings.in
+++ b/scripts/xdg-settings.in
@@ -243,6 +243,23 @@ set_browser_kde()
 }
 
 # }}} KDE
+# {{{ MATE
+get_browser_mate()
+{
+	get_browser_gnome3 "$@"
+}
+
+check_browser_mate()
+{
+	check_browser_gnome3 "$@"
+}
+
+set_browser_mate()
+{
+	set_browser_gnome3 "$@"
+}
+
+# }}} MATE
 # {{{ GNOME
 
 get_browser_gnome()
@@ -738,7 +755,7 @@ fi
 detectDE
 
 case "$DE" in
-    kde|gnome*|xfce)
+    kde|gnome*|mate|xfce)
     dispatch_specific "$@"
     ;;
 
-- 
1.8.0.1

